server/controllers/stock/core.js

```diff
@@ -669,10 +677,21 @@ async function getStockConsumptionAverage(periodId, periodDate, monthAverageCons
   return execStockConsumption;
 }
 
+/**
+ * setDepotPermissionCheck
+ * @param {*} params
+ */
+async function setDepotPermissionCheck(params) {
```

Comment:
This looks like this function is called every time, right?

You can actually make this a lot simpler by just making a custom filter out of it.  Instead of using this function, just put in your custom filter:
```js
// old version:
filters.custom('depot_permission_check', 'd.uuid IN (?)', [params.depot_permission_check]); 

// new version
filters.custom(
  'user_id',
  'd.uuid IN (SELECT depot_uuid FROM depot_permission WHERE user_id = ?)'
); 
```

You then don't need the `setDepotPermissionCheck` function at all!  It should do the same thing.  All you need to do now is check if `user_id` is set or not.

Reply:
If you don't want to use `user_id` (for example, if it clashes with something else), just use another `param` name.